crates/ruff/src/printer.rs

```diff
@@ -264,6 +269,9 @@ impl Printer {
                 self.write_summary_text(writer, diagnostics)?;
             }
             SerializationFormat::Grouped => {
+                if self.flags.intersects(Flags::SHOW_SOURCE) {
+                    warn_user!("`--show-source` with `--output-format=grouped` is deprecated. Source display will not be available for `--output-format=grouped` in future releases.");
+                }
```

Comment:
Is it feasible to warn during CLI parsing instead of during display? I'm not sure if that's much better but then we can easily do things like throw an error if preview mode is enabled (instead of warning).